### PR TITLE
Giving plugin more control over when it should be executed

### PR DESCRIPTION
**Pull request by JKamsker**
Added new Eventlistener to give more control.
Example added

### CODE REVIEW

**Comment by AeonLucid** [comment]
- Solution file looks messed up
- Please stop putting blank lines between the using statements, your old PRs had this too
- Would be great if you could update the description with what this PR accomplishes and an example

**Comment by AeonLucid** [comment]
Also remove the example plugin post build

**Comment by JKamsker** [comment]
> * Solution file looks messed up
What do you mean with that? My visual studio (the most recent preview update) does that, i didnt change it on my own

> * Please stop putting blank lines between the using statements, your old PRs had this too
Im not doing that on my own aswell. Its the FormatDocument of vs.

> * Would be great if you could update the description with what this PR accomplishes and an example
I need control over which events to handle at runtime. I wanna create a javascript bridge, therefor the plugin doesnt know which events to handle at runtime.

**Comment by JKamsker** [comment]
> *Also remove the example plugin post build
I want the plugin to be updated in the plugin dir every time i do smth and not care about copying it every time. What would u suggest?

**Comment by js6pak** [comment]
don't commit shit your vs does, and don't commit stuff you use for development

**Comment by Mm2PL** [comment]
Tip: Look over the files you are about to commit, you don't want to commit things like this. If you want to be extra safe, use the gui equivalent of `git add --patch path/to/file`, as it allows you to add a part of a file.

**Comment by JKamsker** [comment]
> *don't commit shit your vs does, and don't commit stuff you use for development

> *Tip: Look over the files you are about to commit, you don't want to commit things like this. If you want to be extra safe, use the gui equivalent of git add --patch path/to/file, as it allows you to add a part of a file.


So you guys suggest  10 extra steps for doing a simple pr? Thats insane

**Comment by JKamsker** [comment]
That way i should just code in Notepad to be sure that nothing else changes automatically, eh?

**Comment by js6pak** [comment]
Cmon, just change your IDE then, its not more steps, its even less steps, when you are commiting via IDE you can easily check what changes are you commiting, and exlude/include/edit them as you want

**Comment by JKamsker** [comment]
@js6pak But my vs would not accept it, they would just be carried over to the next commit. This sucks!

Adding them to the gitignore wouldnt also be a solution as changes in the sln can be useful

**Comment by Mm2PL** [comment]
> So you guys suggest 10 extra steps for doing a simple pr? Thats insane

In this case other people are doing the steps **for you** and you probably should fix the issues pointed out by AeonLucid.

**Comment by JKamsker** [comment]
Vs changing the sln on its own doesnt suddendly break everything.

**Comment by js6pak** [comment]
Just use your ide ;-;
> Vs changing the sln on its own doesnt suddendly break everything.

It breaks your PR

**Comment by JKamsker** [comment]
> * Please stop putting blank lines between the using statements, your old PRs had this too

@AeonLucid Its an editorconfig setting coming from the project setting. As i already said, im not doing it  on purpose, its autoformat

**Comment by js6pak** [comment]
image
Also unticking one single checkbox is really that hard?

**Comment by JKamsker** [comment]
> Just use your ide ;-;

@js6pak 
What?

> It breaks your PR

in which way

**Comment by JKamsker** [comment]
image
@js6pak Which checkbox?

**Comment by JKamsker** [comment]
As i already said, it would carry over to the next commit and vs would prob redo the change until its finally changed.

Also i dont wanna go through every damn file on each commit

**Comment by js6pak** [comment]
> Also i dont wanna go through every damn file on each commit

;-;

**Comment by js6pak** [comment]
> @js6pak Which checkbox?

I can't understand the language, but you are probably commiting all changes with that button at the top. Not a VS user but you probably have to add files with that little + first

**Comment by AeonLucid** [comment]
You say:
> So you guys suggest 10 extra steps for doing a simple pr? Thats insane

So your solution is to put the burden on the project maintainer?
If I need to do fixes after every PR, that's insane.

**Comment by JKamsker** [comment]
Could you please help me understand in which ways the unintended sln&csproj changes are breaking the project, build or product?

**Comment by AeonLucid** [comment]
It's a change that is not wanted. If every PR changes this we'd have a lot of conflicts.
You can either revert it or the PR won't be merged for sure.

The description of this PR still does not say what the PR does. Quickly reading through the code shows me that it allows you to cancel events from being fired, which is already what `IEventCancelable` does.

File: src/Impostor.sln [hunk 1]
```diff
@@ -163,4 +163,4 @@ Global
 		{4DB56ADD-6D3D-4D0E-A047-9D7E7D40EF99} = {56DD9707-D811-4056-9E2C-8A9CC2479B07}
 		{70F97BB7-12A1-4C7A-B2A5-B962D14B813E} = {36AA9913-E6EA-4A6C-90E6-2FD3CC2E3124}
 	EndGlobalSection
-EndGlobal
+EndGlobal
```
Comment (JKamsker) [inline src/Impostor.sln:166]: IDK WHAT THE FUCK HAPPENS HERE

**Review by JKamsker — COMMENTED** [review]
.

**Comment by JKamsker** [comment]
@AeonLucid  I deleted all the unintended changes. I think ill stop here adding features as i dont feel that im welcomed here anymore.

This pr (as already mentioned) should give the plugin dev the option to receive all events to his liking without constraining it to a specific type at compile time

File: src/Impostor.sln (hunk 1 above)
Comment (reecebenson) [reply src/Impostor.sln:167]: Your formatter is removing the final line of the file

**Pull request merged**